### DISCUSSION TITLE
ERR_SSL_PROTOCOL_ERROR in GitHub Codespace with Next.js TypeScript Project

### DISCUSSION THREAD

**Discussion by nyedr** [community/community#62923] [Codespaces]
### Select Topic Area

Bug

### Body

Hello everyone,

I've recently encountered an issue while working with a Next.js TypeScript project in GitHub Codespace. When trying to run the project and access it via the browser, I'm presented with the following error:

<img src="https://i.postimg.cc/W4Ntsbsy/sslError.png" />

I've tried a few troubleshooting steps, including:
- Accessing the site using HTTP instead of HTTPS.
- Clearing the browser's cache and cookies.
- Disabling browser extensions that might interfere with SSL.
- Ensuring my browser is up-to-date.
- Restarting the Codespace.
- Checking my system's time and date settings.
- Deleting the codespace's domain from chrome net internals.

However, the issue persists. I'm wondering if anyone else has faced a similar issue and if there are any known solutions or workarounds. Additionally, if there's any insight into whether this might be a configuration issue on my end or something related to GitHub Codespaces itself, that would be greatly appreciated.

[Codespace Repository](https://github.com/eidanGar/agile-track)
Operating System: Debian 11
Browser: Google Version 115.0.5790.160 (Official Build) (64-bit)

**Comment by StevenClontz** [comment]
This is an issue for [PreTeXtBook/pretext-cli](https://github.com/pretextbook/pretext-cli) users as well. It seems to only affect folks on certain networks - we had one user show us just now that her browser was erroring as illustrated in the OP, but when we had her switch to her phone's hotspot the error went away.

**Comment by craiglpeters** [comment]
These kinds of connection issues are classically hard to solve as they are so specific to each local environment (OS, client/browser, network, etc). I suggest working though the troubleshooting steps:

[Connection Troubleshooting](https://docs.github.com/en/codespaces/troubleshooting/troubleshooting-your-connection-to-github-codespaces)
[Client Troubleshooting](https://docs.github.com/en/codespaces/troubleshooting/troubleshooting-github-codespaces-clients)

I am unable to reproduce this error as I cannot build the `agile-track` app in the default Codespaces environement.

From the Codespaces Creation Log
> 023-08-16 22:06:09.711Z: Running 'npm run build'...
> 2023-08-16 22:06:09.712Z: 
> 2023-08-16 22:06:10.028Z: 
> 2023-08-16 22:06:10.032Z: > agile-track@0.1.0 build
> 2023-08-16 22:06:10.033Z: > next build
> 2023-08-16 22:06:10.033Z: 
> ...
> 2023-08-16 22:06:10.470Z: 
> 2023-08-16 22:06:10.552Z: - info Creating an optimized production build...
> 2023-08-16 22:06:33.466Z: - info Compiled successfully
> 2023-08-16 22:06:33.472Z: - info Linting and checking validity of types...
> 2023-08-16 22:06:38.206Z: Failed to compile.
> 2023-08-16 22:06:38.207Z: 
> 2023-08-16 22:06:38.209Z: src/app/api/company/new/route.ts
> 2023-08-16 22:06:38.209Z: Type error: Route "src/app/api/company/new/route.ts" does not match the required types of a Next.js Route.
> 2023-08-16 22:06:38.210Z:   "POST" is not a valid Route export field.

**Comment by IgorAguiar-Giro** [comment]
I'm with the same issue working with a Next.JS React application....
